src/OpenTelemetry/Metrics/Exemplar/ExemplarMeasurement.cs

```diff
@@ -12,10 +12,10 @@ namespace OpenTelemetry.Metrics;
 /// <summary>
 /// Represents an Exemplar measurement.
 /// </summary>
-/// <remarks><inheritdoc cref="Exemplar" path="/remarks/para[@experimental-warning='true']"/></remarks>
+/// <remarks><inheritdoc cref="ExemplarReservoir" path="/remarks/para[@experimental-warning='true']"/></remarks>
 /// <typeparam name="T">Measurement type.</typeparam>
 #if NET8_0_OR_GREATER
-[Experimental(DiagnosticDefinitions.ExemplarExperimentalApi, UrlFormat = DiagnosticDefinitions.ExperimentalApiUrlFormat)]
+[Experimental(DiagnosticDefinitions.ExemplarReservoirExperimentalApi, UrlFormat = DiagnosticDefinitions.ExperimentalApiUrlFormat)]
```

Comment:
Do we not need [this](https://github.com/open-telemetry/opentelemetry-dotnet/blob/ac0d1d1fab24db657008e853e2b882096c41a974/src/OpenTelemetry/Metrics/Exemplar/ExemplarMeasurement.cs#L61) exposed?

Reply:
The spec design is you give the `AlignedHistogramBucketExemplarReservoir` the buckets/configuration for the histogram and then when an exemplar is offered you calculate the index and always keep the latest update:

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#alignedhistogrambucketexemplarreservoir

This `internal int ExplicitBucketHistogramBucketIndex` allows the SDK to pass it in so we don't have to do the calculation twice.

Should it be exposed?

It isn't part of the spec so I don't know if we can/should expose it. Chatting with @alanwest, it does seem like a useful thing to have available to custom reservoirs. The spec could have "MAY" language about exposing it. But it isn't a blocker for the current plan for the first stable release. In general, we (me, @alanwest, @cijothomas) have a lot of questions/concerns about the reservoir design which is why I think it is a good idea to keep it in preview initially.

Reply:
Sounds good to me 👍 . I asked because of this reason only - `it does seem like a useful thing to have available to custom reservoirs`. The [AlignedHistogramBucketExemplarReservoir](https://github.com/open-telemetry/opentelemetry-dotnet/blob/8864a054b622d9a8e6aa8ef72b5df69b7594e0bb/src/OpenTelemetry/Metrics/Exemplar/AlignedHistogramBucketExemplarReservoir.cs#L33) uses this so I thought someone writing the custom reservoir for histogram may also need this information.